log-manager/src/main/java/io/airlift/log/BufferedHandler.java

```diff
@@ -368,6 +390,39 @@ public void flush()
         flushMessageOutputSafe();
     }
 
+    /**
+     * Requests and awaits all currently records present in queue at the time of the call be written to the output
+     * and flushed fully before returning. If the handler is closed, the queue is full (and therefore the flush request can't be enqueued),
+     * or processing the flush request takes longer than the provided timeout, this method will return <code>false</code>,
+     * otherwise it returns <code>true</code>.
+     *
+     * @param timeout The maximum amount of time to wait for the flush request to process
+     * @param unit The {@link TimeUnit} for the flush request timeout argument
+     * @return true if the flush completed successfully within the timeout, false otherwise
+     */
+    public boolean flushFullyAndAwait(long timeout, TimeUnit unit)
+    {
+        synchronized (fullFlushMutex) {
+            queueDrainLock.lock();
+            try {
+                if (inputClosed.get() || terminalMessageDequeued) {
+                    return false;
+                }
+                if (!queue.offer(FLUSH_MESSAGE)) {
+                    return false;
+                }
+                return recordsFlushed.await(timeout, unit);
```

Comment:
This needs to awake the background thread before awaiting the condition, will fix shortly